src/solver/infeasible-problem-analysis/include/antares/solver/infeasible-problem-analysis/watched-constraints.h

```diff
@@ -0,0 +1,104 @@
+
+#pragma once
+
+#include <functional>
+#include <map>
+#include <memory>
+#include <regex>
+#include <string>
+#include <vector>
+
+namespace Antares::Optimization
+{
+class WatchedConstraint
+{
+public:
+    WatchedConstraint(std::string constraintName);
```

Comment:
```suggestion
    WatchedConstraint(const std::string& constraintName);
```
avoid copies, or use `std::move` later.

Reply:
Done